components/style/properties/shorthand/position.mako.rs

```diff
@@ -512,7 +510,7 @@
             if *self.grid_template_areas != Either::Second(None_) ||
                (*self.grid_template_rows != GridTemplateComponent::None &&
                    *self.grid_template_columns != GridTemplateComponent::None) ||
-               self.is_initial() {
+               self.is_grid_template() {
```

Comment:
This is not because of gecko actually. This is because of strange syntax of grid property. You have to do some additional checks for it to be able to serialize it correctly.